pkg/controllers/clusteroperator_controller.go

```diff
@@ -189,18 +188,18 @@ func (r *CloudOperatorReconciler) SetupWithManager(mgr ctrl.Manager) error {
 
 	build := ctrl.NewControllerManagedBy(mgr).
 		For(&configv1.ClusterOperator{}, builder.WithPredicates(clusterOperatorPredicates())).
-		Watches(&source.Kind{Type: &configv1.Infrastructure{}},
+		Watches(&configv1.Infrastructure{},
 			handler.EnqueueRequestsFromMapFunc(toClusterOperator),
 			builder.WithPredicates(infrastructurePredicates())).
-		Watches(&source.Kind{Type: &configv1.FeatureGate{}},
+		Watches(&configv1.FeatureGate{},
 			handler.EnqueueRequestsFromMapFunc(toClusterOperator),
 			builder.WithPredicates(featureGatePredicates())).
-		Watches(&source.Kind{Type: &operatorv1.KubeControllerManager{}},
+		Watches(&operatorv1.KubeControllerManager{},
 			handler.EnqueueRequestsFromMapFunc(toClusterOperator),
 			builder.WithPredicates(kcmPredicates())).
-		Watches(&source.Channel{Source: watcher.EventStream()}, handler.EnqueueRequestsFromMapFunc(toClusterOperator)).
-		Watches(&source.Kind{Type: &corev1.ConfigMap{}}, handler.EnqueueRequestsFromMapFunc(toClusterOperator)).
-		Watches(&source.Kind{Type: &corev1.Secret{}}, handler.EnqueueRequestsFromMapFunc(toClusterOperator))
+		//Watches(&source.Channel{Source: watcher.EventStream()}, handler.EnqueueRequestsFromMapFunc(toClusterOperator)).
```

Comment:
Commented out?

Reply:
Tbh, I don't really know what to do with this channel. Don't you know about a way I could keep the same behavior with the channel somehow being an Object?

Reply:
You can change it to use [this](https://github.com/kubernetes-sigs/controller-runtime/blob/e7b94074ad3814832bbd5f254bb9c5c16a286a35/pkg/builder/controller.go#L174-L187), `WatchesRawSource`, I'm going to trust that we knew what we were doing when we added this event stream

Reply:
Great, thanks!